ShareIt

Enfurecid0 opened the pull request.

Enfurecid0 committed:
Создать DTO-объекты, мапперы, сервисы, контроллеры.

VadimZharkov commented on src/main/java/ru/practicum/shareit/exceptions/ErrorHandler.java:
```diff
@@ -0,0 +1,31 @@
+    @ExceptionHandler
+    @ResponseStatus(HttpStatus.NOT_FOUND)
+    public ErrorResponse handleNotFoundException(NotFoundException e) {
+        return new ErrorResponse(e.getMessage());
```
Хорощо бы еще добавить логирование в методы обработчика ошибок. Так будет проще разбираться с ошибками.

VadimZharkov commented on src/main/java/ru/practicum/shareit/exceptions/ErrorHandler.java:
```diff
@@ -0,0 +1,31 @@
+    public ErrorResponse handleEmailValidException(ValidException e) {
+        return new ErrorResponse(e.getMessage());
+    }
+
```
Нужно еще добавить обработку всех оставшихся ошибок
```
 @ExceptionHandler
 @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
 public ErrorResponse handleEmailValidException(Throwable e) 
```

VadimZharkov commented on src/main/java/ru/practicum/shareit/item/controller/ItemController.java:
```diff
@@ -0,0 +1,60 @@
+    }
+
+    @GetMapping("/{itemId}")
+    @ResponseStatus(HttpStatus.OK)
```
`@ResponseStatus(HttpStatus.OK) `можно не указывать, т.к. он является статусом по умолчанию у методов класса `@RestController`

VadimZharkov commented on src/main/java/ru/practicum/shareit/item/repository/InMemoryItemStorage.java:
```diff
@@ -0,0 +1,100 @@
+
+    @Override
+    public Item createItem(Long userId, Item item) {
+        if (userStorage.getUser(userId) == null) {
```
Эти все проверки связаны с бизнес логикой и поэтому должны располагаться в классе-сервисе, т.к бизнес-логика - это зона ответственности класса-сервиса. Класс-хранилище он достаточно примитивен, просто хранит все что пришло от класса-сервиса.

VadimZharkov reviewed (changes requested):
Хорошая работа, аккуратный код. Посмотри, что можно улучшить. А также нужно перенести всю бизнес-логику из класса-хранилища в класс-сервис.

Enfurecid0 committed:
Добавлено логирование в ErrorHandler.
Добавлено обработка всех оставшихся ошибок.
Проверки перенесены в класс-сервис.
